FinanceManager.Core/Entities/Accounts/BankAccount.cs

```diff
@@ -33,6 +17,39 @@ public BankAccount(int id, string name, AccountType accountType) : base(id, name
             AccountType = accountType;
             Entries = new List<BankAccountEntry>();
         }
+
+        public void Add(AddBankEntryDto entry)
+        {
+            Entries ??= new List<BankAccountEntry>();
+            var alredyExistingEntry = Entries.FirstOrDefault(x => x.PostingDate == entry.PostingDate && x.ValueChange == entry.ValueChange);
+            if (alredyExistingEntry is not null)
+            {
+                throw new Exception($"WARNING - Entry already exist, can not be added: Id:{alredyExistingEntry.Id}, Posting date{alredyExistingEntry.PostingDate}, " +
+                    $"Value change {alredyExistingEntry.ValueChange}");
+            }
+
+            var previousEntry = Entries.GetPrevious(entry.PostingDate).FirstOrDefault();
+            var index = -1;
+
+            if (previousEntry is not null)
+                index = Entries.IndexOf(previousEntry);
+
+            BankAccountEntry newEntry = null;
+            if (index == -1)
+            {
+                index = Entries.Count();
+                newEntry = new BankAccountEntry(GetNextFreeId(), entry.PostingDate, entry.ValueChange, entry.ValueChange);
+                Entries.Add(newEntry);
+                index -= 1;
+            }
+            else
+            {
+                newEntry = new BankAccountEntry(GetNextFreeId(), entry.PostingDate, entry.ValueChange, entry.ValueChange);
+                Entries.Insert(index, newEntry);
+            }
+
+            RecalculateEntryValues(index);
```

Comment:
why this method exisit?

Reply:
Let say you have 3 transactions

- `02-01-2000; +10 pln; 10 pln`
- `03-01-2000; +10 pln; 20 pln`

you add new one
`01-01-2000; +10 pln; 10 pln`

without `RecalculateEntryValues()` you would end up with following ledger 


- `01-01-2000; +10 pln; 10 pln`
- `02-01-2000; +10 pln; 10 pln`
- `03-01-2000; +10 pln; 20 pln`

`RecalculateEntryValues()` will recalculate balances and make it right

- `01-01-2000; +10 pln; 10 pln`
- `02-01-2000; +10 pln; 20 pln`
- `03-01-2000; +10 pln; 30 pln`